src/acquisition/covidcast/database.py

```diff
@@ -95,106 +110,79 @@ def disconnect(self, commit):
-    insert_or_update_sql = f'''
-      INSERT INTO `covidcast`
-        (`source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`,
-        `value_updated_timestamp`, `value`, `stderr`, `sample_size`, `direction_updated_timestamp`, `direction`,
-        `issue`, `lag`, `is_latest_issue`, `is_wip`, `missing_value`, `missing_stderr`, `missing_sample_size`)
-      SELECT * FROM `{tmp_table_name}`
-      ON DUPLICATE KEY UPDATE
-        `value_updated_timestamp` = VALUES(`value_updated_timestamp`),
-        `value` = VALUES(`value`),
-        `stderr` = VALUES(`stderr`),
-        `sample_size` = VALUES(`sample_size`)
-    '''
-    zero_is_latest_issue_sql = f'''
-      UPDATE
-      (
-        SELECT DISTINCT `source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`
-        FROM `{tmp_table_name}`
-      ) AS TMP
-      LEFT JOIN `covidcast`
-      USING (`source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`)
-      SET `is_latest_issue`=0
+    # all load table entries are already marked "is_latest_issue".
+    # if an entry in the load table is NOT in the latest table, it is clearly now the latest value for that key (so we do nothing (thanks to INNER join)).
+    # if an entry *IS* in both load and latest tables, but latest table issue is newer, unmark is_latest_issue in load.
+    fix_is_latest_issue_sql = f'''
+        UPDATE 
+            `{self.load_table}` JOIN `{self.latest_view}`
+                USING (`source`, `signal`, `geo_type`, `geo_value`, `time_type`, `time_value`) 
+            SET `{self.load_table}`.`is_latest_issue`=0 
+            WHERE `{self.load_table}`.`issue` < `{self.latest_view}`.`issue`
+                  AND `process_status` = '{PROCESS_STATUS.INSERTING}'
     '''
-    set_is_latest_issue_sql = f'''
-      UPDATE 
-      (
-        SELECT `source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`, MAX(`issue`) AS `issue`
-        FROM
-        (
-          SELECT DISTINCT `source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value` 
-          FROM `{tmp_table_name}`
-        ) AS TMP
-        LEFT JOIN `covidcast`
-        USING (`source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`)
-        GROUP BY `source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`
-      ) AS TMP
-      LEFT JOIN `covidcast`
-      USING (`source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`, `issue`)
-      SET `is_latest_issue`=1        
+
+    update_status_sql = f'''
+        UPDATE `{self.load_table}`
+            SET `process_status` = '{PROCESS_STATUS.LOADED}'
+            WHERE `process_status` = '{PROCESS_STATUS.INSERTING}'
     '''
 
+    if 0 != self.count_insertstatus_rows():
+      # TODO: determine if this should be fatal?!
+      logger = get_structured_logger("insert_or_update_batch")
+      logger.warn("Non-zero count in the load table!!!  This indicates scheduling of acqusition and dbjobs may be out of sync.")
```

Comment:
also, kinda naive question: i never see `insert_or_update_batch` and `run_db_jobs` run separately. if we expect them to be so tightly coupled, why not have `csv_to_database.py` call `run_db_jobs` in main and not rely on outside script scheduling to keep sync?

Reply:
HARD AGREE

Reply:
they are run seperately (at least in this branch) : see https://github.com/cmu-delphi/delphi-epidata/blob/v4-schema-revisions-release-prep-prep/src/acquisition/covidcast/dbjobs_runner.py which was intended to be run periodically by our automation system.

but this is an older design decision.  branch `v4-srrpp-migrations` does away with the independence of those 2 methods, except that `dbjobs` is kept separate for doing our final migration from v3.